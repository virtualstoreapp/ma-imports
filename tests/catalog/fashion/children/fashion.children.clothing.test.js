const {
  setupDesktop,
  setupMobile,
} = require('../../../utils/catalogCommon');

const {
  selectSweatshirtSetChildren,
} = require('../../../utils/catalogActions');

describe('Fashion', () => {
    describe('Children', () => {
        describe('Clothing', () => {

            describe('Desktop View', () => {
                beforeEach(async () => {
                    await setupDesktop();
                });

                describe('Sweatshirts Set', () => {
                    it('renders final subcategory "Conjuntos Moletom Infantil" correctly on desktop', async () => {
                        await selectSweatshirtSetChildren();
                    });
                });

            });

            describe('Mobile View', () => {
                beforeEach(async () => {
                    await setupMobile();
                });

                describe('Sweatshirts Set', () => {
                    it('renders final subcategory "Conjuntos Moletom Infantil" correctly on mobile', async () => {
                        await selectSweatshirtSetChildren();
                    });
                });
            });
        });
    });
});