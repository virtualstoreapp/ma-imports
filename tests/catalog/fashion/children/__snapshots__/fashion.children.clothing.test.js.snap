// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Fashion Children Clothing Desktop View Sweatshirts Set renders final subcategory "Conjuntos Moletom Infantil" correctly on desktop 1`] = `
"
  <header>
    <div class="header-logo">
      <img src="images/logo.jpeg" alt="M/A Imports Logo">
    </div>
  </header>

  <!-- Navigation -->
  <nav aria-label="Main Navigation">
    <button id="menu-toggle" aria-expanded="false" aria-controls="nav-menu">☰</button>
    <ul id="nav-menu">
      <li class="menu-item">
        <button data-category="all">Início</button>
      </li>
      <li class="menu-item has-submenu">
        <button data-category="fashion-category" aria-expanded="false">Moda</button>
        <ul class="submenu" style="display: none;">
          <li class="menu-item has-submenu">
            <button data-category="man-subcategory" aria-expanded="false">Masculino</button>
            <ul class="submenu" style="display: none;">
              <li class="menu-item has-submenu">
                <button data-category="accessories-man-subcategory" aria-expanded="false">Acessórios</button>
                <ul class="submenu" style="display: none;">
                  <li class="menu-item">
                    <button data-category="caps-man">Bonés</button>
                    <button data-category="wallets-man">Carteiras</button>
                    <button data-category="belts-man">Cintos</button>
                  </li>
                </ul>
              </li>
              <li class="menu-item has-submenu">
                <button data-category="shoes-man-subcategory" aria-expanded="false">Calçados</button>
                <ul class="submenu" style="display: none;">
                  <li class="menu-item">
                    <button data-category="slippers-man">Chinelos</button>
                    <button data-category="socks-man">Meias</button>
                    <button data-category="shoes-man">Tênis</button>
                  </li>
                </ul>
              </li>
              <li class="menu-item has-submenu">
                <button data-category="clothing-man-subcategory" aria-expanded="false">Roupas</button>
                <ul class="submenu" style="display: none;">
                  <li class="menu-item has-submenu">
                    <button data-category="shorts-man-subcategory" aria-expanded="false">Bermudas</button>
                    <ul class="submenu" style="display: none;">
                      <li class="menu-item">
                        <button data-category="shorts-basic-man">Básica</button>
                        <button data-category="shorts-jeans-man">Jeans</button>
                        <button data-category="shorts-sweatshorts-man">Moletom</button>
                        <button data-category="shorts-tactel-man">Tactel</button>
                      </li>
                    </ul>
                    <button data-category="sweatshirts-man-subcategory" aria-expanded="false">Blusas</button>
                    <ul class="submenu" style="display: none;">
                      <li class="menu-item">
                        <button data-category="sweatshirts-man">Blusas</button>
                      </li>
                    </ul>
                    <button data-category="pants-man-subcategory" aria-expanded="false">Calças</button>
                    <ul class="submenu" style="display: none;">
                      <li class="menu-item">
                        <button data-category="pants-jeans-man">Jeans</button>
                        <button data-category="pants-sweatpants-man">Moletom</button>
                      </li>
                    </ul>
                    <button data-category="tshirts-man-subcategory" aria-expanded="false">Camisetas</button>
                    <ul class="submenu" style="display: none;">
                      <li class="menu-item">
                        <button data-category="tshirts-casual-man">Casuais</button>
                        <button data-category="tshirts-dryfit-man">Dry Fit</button>
                        <button data-category="tshirts-polo-man">Polo</button>
                        <button data-category="dress-shirts-man">Sociais</button>
                      </li>
                    </ul>
                    <button data-category="underwear-man-subcategory" aria-expanded="false">Cuecas</button>
                    <button data-category="tank-top-man-subcategory" aria-expanded="false">Regatas</button>
                    <ul class="submenu" style="display: none;">
                      <li class="menu-item">
                        <button data-category="tank-top-casual-man">Casuais</button>
                        <button data-category="tank-top-dryfit-man">Dry Fit</button>
                      </li>
                    </ul>
                  </li>
                </ul>
              </li>
            </ul>
          </li>
          <li class="menu-item has-submenu">
            <button data-category="woman-subcategory" aria-expanded="false">Feminino</button>
            <ul class="submenu" style="display: none;">
              <li class="menu-item has-submenu">
                <button data-category="clothing-woman-subcategory" aria-expanded="false">Roupas</button>
                <ul class="submenu" style="display: none;">
                  <li class="menu-item has-submenu">
                    <button data-category="sweatshirts-woman-subcategory" aria-expanded="false">Blusas</button>
                    <ul class="submenu" style="display: none;">
                      <li class="menu-item">
                        <button data-category="sweatshirts-woman">Blusas</button>
                      </li>
                    </ul>
                  </li>
                  <li class="menu-item has-submenu">
                    <button data-category="shorts-woman-subcategory" aria-expanded="false">Bermudas</button>
                    <ul class="submenu" style="display: none;">
                      <li class="menu-item">
                        <button data-category="shorts-jeans-woman">Jeans</button>
                      </li>
                    </ul>
                  </li>
                  <li class="menu-item has-submenu">
                    <button data-category="pants-woman-subcategory" aria-expanded="false">Calças</button>
                    <ul class="submenu" style="display: none;">
                      <li class="menu-item">
                        <button data-category="pants-jeans-woman">Jeans</button>
                      </li>
                    </ul>
                  </li>
                  <li class="menu-item has-submenu">
                    <button data-category="fitness-woman-subcategory" aria-expanded="false">Fitness</button>
                    <ul class="submenu" style="display: none;">
                      <li class="menu-item">
                        <button data-category="fitness-legging-woman">Legging</button>
                        <button data-category="fitness-top-woman">Top</button>
                      </li>
                    </ul>
                  </li>
                </ul>
              </li>
            </ul>
          </li>
          <li class="menu-item has-submenu">
            <button data-category="children-subcategory" aria-expanded="false">Infantil</button>
            <ul class="submenu" style="display: none;">
              <li class="menu-item has-submenu">
                <button data-category="clothing-children-subcategory" aria-expanded="false">Roupas</button>
                <ul class="submenu" style="display: none;">
                  <li class="menu-item has-submenu">
                    <button data-category="sets-children-subcategory" aria-expanded="false">Conjuntos</button>
                    <ul class="submenu" style="display: none;">
                      <li class="menu-item">
                        <button data-category="sweatshirts-set-children">Moletom</button>
                      </li>
                    </ul>
                  </li>
                </ul>
              </li>
            </ul>
          </li>
        </ul>
      </li>
    </ul>
  </nav>

  <!-- Main Content -->
  <main>
    <h2 id="category-heading">Conjuntos Moletom Infantil</h2>
    <ul id="product-list"><li class="product-item" style="position: relative;">
          <img src="images/children/sweatshirts/1007251301-nike.jpeg" alt="[1007251301] Nike">
          <div class="product-details">
            <h3>[1007251301] Nike</h3>
            <p class="description">Conjunto Moleton Infantil, Cor: Preta</p>
            <span class="size">Tamanho: 8</span>
            <span class="price">R$&nbsp;199,00</span>
          </div>
        </li><li class="product-item" style="position: relative;">
          <img src="images/children/sweatshirts/1007251302-nike.jpeg" alt="[1007251302] Nike">
          <div class="product-details">
            <h3>[1007251302] Nike</h3>
            <p class="description">Conjunto Moleton Infantil, Cor: Preta</p>
            <span class="size">Tamanho: 12</span>
            <span class="price">R$&nbsp;199,00</span>
          </div>
        </li><li class="product-item" style="position: relative;">
          <img src="images/children/sweatshirts/1007251303-nike.jpeg" alt="[1007251303] Nike">
          <div class="product-details">
            <h3>[1007251303] Nike</h3>
            <p class="description">Conjunto Moleton Infantil, Cor: Azul Marinho</p>
            <span class="size">Tamanho: 16</span>
            <span class="price">R$&nbsp;199,00</span>
          </div>
        </li><li class="product-item" style="position: relative;">
          <img src="images/children/sweatshirts/1007251304-jordan.jpeg" alt="[1007251304] Jordan">
          <div class="product-details">
            <h3>[1007251304] Jordan</h3>
            <p class="description">Conjunto Moleton Infantil, Cor: Preta</p>
            <span class="size">Tamanho: 14</span>
            <span class="price">R$&nbsp;199,00</span>
          </div>
        </li><li class="product-item" style="position: relative;">
          <img src="images/children/sweatshirts/0207251346-lacoste.jpeg" alt="[0207251346] Lacoste">
          <div class="product-details">
            <h3>[0207251346] Lacoste</h3>
            <p class="description">Conjunto Moleton Infantil, Cor: Branca</p>
            <span class="size">Tamanho: 10</span>
            <span class="price">R$&nbsp;179,00</span>
          </div>
        </li></ul>
  </main>

  <footer>
    <p>Versão: 5.6.0 - 0.61.234</p>
  </footer>

  <script src="scripts.js"></script>


<div id="product-modal">
      <div id="modal-content">
        <button id="modal-close">X</button>
        <div id="modal-image-container" class="modal-image-container">
          <img id="modal-image" src="" alt="">
        </div>
        <div id="modal-controls">
          <button id="prev-image">&lt;</button>
          <button id="zoom-out">-</button>
          <button id="copy-product-id">ID</button>
          <button id="zoom-in">+</button>
          <button id="next-image">&gt;</button>
          <button id="buy-product">Comprar</button>
        </div>
      </div>
    </div>"
`;

exports[`Fashion Children Clothing Mobile View Sweatshirts Set renders final subcategory "Conjuntos Moletom Infantil" correctly on mobile 1`] = `
"
  <header>
    <div class="header-logo">
      <img src="images/logo.jpeg" alt="M/A Imports Logo">
    </div>
  </header>

  <!-- Navigation -->
  <nav aria-label="Main Navigation" class="">
    <button id="menu-toggle" aria-expanded="false" aria-controls="nav-menu">☰</button>
    <ul id="nav-menu">
      <li class="menu-item">
        <button data-category="all">Início</button>
      </li>
      <li class="menu-item has-submenu">
        <button data-category="fashion-category" aria-expanded="false">Moda</button>
        <ul class="submenu" style="display: none;">
          <li class="menu-item has-submenu">
            <button data-category="man-subcategory" aria-expanded="false">Masculino</button>
            <ul class="submenu" style="display: none;">
              <li class="menu-item has-submenu">
                <button data-category="accessories-man-subcategory" aria-expanded="false">Acessórios</button>
                <ul class="submenu" style="display: none;">
                  <li class="menu-item">
                    <button data-category="caps-man">Bonés</button>
                    <button data-category="wallets-man">Carteiras</button>
                    <button data-category="belts-man">Cintos</button>
                  </li>
                </ul>
              </li>
              <li class="menu-item has-submenu">
                <button data-category="shoes-man-subcategory" aria-expanded="false">Calçados</button>
                <ul class="submenu" style="display: none;">
                  <li class="menu-item">
                    <button data-category="slippers-man">Chinelos</button>
                    <button data-category="socks-man">Meias</button>
                    <button data-category="shoes-man">Tênis</button>
                  </li>
                </ul>
              </li>
              <li class="menu-item has-submenu">
                <button data-category="clothing-man-subcategory" aria-expanded="false">Roupas</button>
                <ul class="submenu" style="display: none;">
                  <li class="menu-item has-submenu">
                    <button data-category="shorts-man-subcategory" aria-expanded="false">Bermudas</button>
                    <ul class="submenu" style="display: none;">
                      <li class="menu-item">
                        <button data-category="shorts-basic-man">Básica</button>
                        <button data-category="shorts-jeans-man">Jeans</button>
                        <button data-category="shorts-sweatshorts-man">Moletom</button>
                        <button data-category="shorts-tactel-man">Tactel</button>
                      </li>
                    </ul>
                    <button data-category="sweatshirts-man-subcategory" aria-expanded="false">Blusas</button>
                    <ul class="submenu" style="display: none;">
                      <li class="menu-item">
                        <button data-category="sweatshirts-man">Blusas</button>
                      </li>
                    </ul>
                    <button data-category="pants-man-subcategory" aria-expanded="false">Calças</button>
                    <ul class="submenu" style="display: none;">
                      <li class="menu-item">
                        <button data-category="pants-jeans-man">Jeans</button>
                        <button data-category="pants-sweatpants-man">Moletom</button>
                      </li>
                    </ul>
                    <button data-category="tshirts-man-subcategory" aria-expanded="false">Camisetas</button>
                    <ul class="submenu" style="display: none;">
                      <li class="menu-item">
                        <button data-category="tshirts-casual-man">Casuais</button>
                        <button data-category="tshirts-dryfit-man">Dry Fit</button>
                        <button data-category="tshirts-polo-man">Polo</button>
                        <button data-category="dress-shirts-man">Sociais</button>
                      </li>
                    </ul>
                    <button data-category="underwear-man-subcategory" aria-expanded="false">Cuecas</button>
                    <button data-category="tank-top-man-subcategory" aria-expanded="false">Regatas</button>
                    <ul class="submenu" style="display: none;">
                      <li class="menu-item">
                        <button data-category="tank-top-casual-man">Casuais</button>
                        <button data-category="tank-top-dryfit-man">Dry Fit</button>
                      </li>
                    </ul>
                  </li>
                </ul>
              </li>
            </ul>
          </li>
          <li class="menu-item has-submenu">
            <button data-category="woman-subcategory" aria-expanded="false">Feminino</button>
            <ul class="submenu" style="display: none;">
              <li class="menu-item has-submenu">
                <button data-category="clothing-woman-subcategory" aria-expanded="false">Roupas</button>
                <ul class="submenu" style="display: none;">
                  <li class="menu-item has-submenu">
                    <button data-category="sweatshirts-woman-subcategory" aria-expanded="false">Blusas</button>
                    <ul class="submenu" style="display: none;">
                      <li class="menu-item">
                        <button data-category="sweatshirts-woman">Blusas</button>
                      </li>
                    </ul>
                  </li>
                  <li class="menu-item has-submenu">
                    <button data-category="shorts-woman-subcategory" aria-expanded="false">Bermudas</button>
                    <ul class="submenu" style="display: none;">
                      <li class="menu-item">
                        <button data-category="shorts-jeans-woman">Jeans</button>
                      </li>
                    </ul>
                  </li>
                  <li class="menu-item has-submenu">
                    <button data-category="pants-woman-subcategory" aria-expanded="false">Calças</button>
                    <ul class="submenu" style="display: none;">
                      <li class="menu-item">
                        <button data-category="pants-jeans-woman">Jeans</button>
                      </li>
                    </ul>
                  </li>
                  <li class="menu-item has-submenu">
                    <button data-category="fitness-woman-subcategory" aria-expanded="false">Fitness</button>
                    <ul class="submenu" style="display: none;">
                      <li class="menu-item">
                        <button data-category="fitness-legging-woman">Legging</button>
                        <button data-category="fitness-top-woman">Top</button>
                      </li>
                    </ul>
                  </li>
                </ul>
              </li>
            </ul>
          </li>
          <li class="menu-item has-submenu">
            <button data-category="children-subcategory" aria-expanded="false">Infantil</button>
            <ul class="submenu" style="display: none;">
              <li class="menu-item has-submenu">
                <button data-category="clothing-children-subcategory" aria-expanded="false">Roupas</button>
                <ul class="submenu" style="display: none;">
                  <li class="menu-item has-submenu">
                    <button data-category="sets-children-subcategory" aria-expanded="false">Conjuntos</button>
                    <ul class="submenu" style="display: none;">
                      <li class="menu-item">
                        <button data-category="sweatshirts-set-children">Moletom</button>
                      </li>
                    </ul>
                  </li>
                </ul>
              </li>
            </ul>
          </li>
        </ul>
      </li>
    </ul>
  </nav>

  <!-- Main Content -->
  <main>
    <h2 id="category-heading">Conjuntos Moletom Infantil</h2>
    <ul id="product-list"><li class="product-item" style="position: relative;">
          <img src="images/children/sweatshirts/1007251301-nike.jpeg" alt="[1007251301] Nike">
          <div class="product-details">
            <h3>[1007251301] Nike</h3>
            <p class="description">Conjunto Moleton Infantil, Cor: Preta</p>
            <span class="size">Tamanho: 8</span>
            <span class="price">R$&nbsp;199,00</span>
          </div>
        </li><li class="product-item" style="position: relative;">
          <img src="images/children/sweatshirts/1007251302-nike.jpeg" alt="[1007251302] Nike">
          <div class="product-details">
            <h3>[1007251302] Nike</h3>
            <p class="description">Conjunto Moleton Infantil, Cor: Preta</p>
            <span class="size">Tamanho: 12</span>
            <span class="price">R$&nbsp;199,00</span>
          </div>
        </li><li class="product-item" style="position: relative;">
          <img src="images/children/sweatshirts/1007251303-nike.jpeg" alt="[1007251303] Nike">
          <div class="product-details">
            <h3>[1007251303] Nike</h3>
            <p class="description">Conjunto Moleton Infantil, Cor: Azul Marinho</p>
            <span class="size">Tamanho: 16</span>
            <span class="price">R$&nbsp;199,00</span>
          </div>
        </li><li class="product-item" style="position: relative;">
          <img src="images/children/sweatshirts/1007251304-jordan.jpeg" alt="[1007251304] Jordan">
          <div class="product-details">
            <h3>[1007251304] Jordan</h3>
            <p class="description">Conjunto Moleton Infantil, Cor: Preta</p>
            <span class="size">Tamanho: 14</span>
            <span class="price">R$&nbsp;199,00</span>
          </div>
        </li><li class="product-item" style="position: relative;">
          <img src="images/children/sweatshirts/0207251346-lacoste.jpeg" alt="[0207251346] Lacoste">
          <div class="product-details">
            <h3>[0207251346] Lacoste</h3>
            <p class="description">Conjunto Moleton Infantil, Cor: Branca</p>
            <span class="size">Tamanho: 10</span>
            <span class="price">R$&nbsp;179,00</span>
          </div>
        </li></ul>
  </main>

  <footer>
    <p>Versão: 5.6.0 - 0.61.234</p>
  </footer>

  <script src="scripts.js"></script>


<div id="product-modal">
      <div id="modal-content">
        <button id="modal-close">X</button>
        <div id="modal-image-container" class="modal-image-container">
          <img id="modal-image" src="" alt="">
        </div>
        <div id="modal-controls">
          <button id="prev-image">&lt;</button>
          <button id="zoom-out">-</button>
          <button id="copy-product-id">ID</button>
          <button id="zoom-in">+</button>
          <button id="next-image">&gt;</button>
          <button id="buy-product">Comprar</button>
        </div>
      </div>
    </div>"
`;
